src/aiortc/rtcpeerconnection.py

```diff
@@ -928,6 +928,8 @@ async def setRemoteDescription(
                     iceTransport._role_set = True
 
                 # set DTLS role
+                if description.type == "offer" and media.dtls.role == "client":
+                    dtlsTransport._set_role(role="server")
                 if description.type == "answer":
                     dtlsTransport._set_role(
                         role="server" if media.dtls.role == "client" else "client"
```

Comment:
Btw what role do we end up with when none of these conditionals match?

Reply:
`auto` which I *hope* means "determined by who wins ICE"

Reply:
I'm not sure I understand why we are only handling `media.dtls.role == "client"`?

Reply:
This is the handling of offers so the existing code assumes actpass. Lets see if I can prove that by adding more tests because who does not like more tests ;-)

Reply:
As I suspected... the current code assumes role=auto in SRD. Which is not quite correct since auto means "determined by DTLS" whereas when we parse actpass we should be setting this to "client" explicitly. This is currently done in createAnswer which is a bit confusing but ok.